view/frontend/templates/page/analytics.phtml

```diff
@@ -4,11 +4,15 @@
  */
 $event           = $block->getEvent();
 $url             = $block->getLibraryUrl();
-$addToCartEvents = $block->getCatalogSession()->getData('metrilo_add_to_cart', true);
+$addToCartEvents = $block->getCartEvents()->getSessionEvents('metrilo_add_to_cart');
 ?>
 <script type="text/javascript">
     require(['<?php echo $url; ?>'], function(){
-        <?php if($addToCartEvents){ $block->cartEventsLoop($addToCartEvents, 'addToCart'); } ?>
+        <?php if($addToCartEvents){
```

Comment:
Why do you need this check? Wouldn't `addToCartEvents` be just `[]` and hence the forEach will not run?

Reply:
Agree, foreach will not run on empty array. Removed.